homeassistant/components/xiaomi_miio/alarm_control_panel.py

```diff
@@ -123,9 +123,10 @@ async def async_update(self):
         try:
             state = await self.hass.async_add_executor_job(self._gateway.alarm.status)
         except DeviceException as ex:
-            self._available = False
-            _LOGGER.error("Got exception while fetching the state: %s", ex)
-            return
+            if self._available:
+                self._available = False
+                _LOGGER.error("Got exception while fetching the state: %s", ex)
+                return
```

Comment:
Wouldn't you still want to exit even if it was already unavailable?

Reply:
Indeed, fixed those cases.